pkg/asset/installconfig/azure/validation.go

```diff
@@ -161,6 +163,30 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 	return allErrs
 }
 
+// validateDiskEncryption checks if the specified disk encryption is valid.
+func validateDiskEncryption(client API, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if ic.Platform.Azure == nil || ic.Platform.Azure.DefaultMachinePlatform == nil {
+		return allErrs
+	}
+
+	encryptionAtHost := ic.Platform.Azure.DefaultMachinePlatform.EncryptionAtHost
+	if ic.Azure.CloudName == aztypes.StackCloud && encryptionAtHost == true {
+		return append(allErrs, field.Invalid(field.NewPath("encryptionAtHost"), encryptionAtHost, "encryption at host is not supported not this platform"))
+	}
+
+	encryptionType := ic.Platform.Azure.DefaultMachinePlatform.OSDisk.EncryptionType
+	if ic.Azure.CloudName == aztypes.StackCloud && encryptionType != "" {
+		return append(allErrs, field.Invalid(field.NewPath("encryptionType"), encryptionType, "customer managed keys encryption is not supported on this platform"))
+	}
```

Comment:
I know very little about azure stack, but I was under the impression that it has the same set of features as the public cloud. Am I wrong?

Or do we just want to limit the scope of this feature to non-Azure Stack clouds for now?

Reply:
It's been decided (for now), to just throw an error if these fields are set for Azure stack.